oar/cli/cmd_image_signed_check.py

```diff
@@ -6,41 +6,79 @@
 from bs4 import BeautifulSoup
 from oar.core.const import *
 from oar.core.worksheet_mgr import WorksheetManager
-
+import time
 
 logger = logging.getLogger(__name__)
 
+
+# 1. go to main URL and grab all links that matches the build type
+# 2. construct a complete URL,
+def get_image_digest(url, current_try=0):
+    print(f"Getting digest from {url}")
+    res = requests.get(url)
+    if not res.ok:
+        logger.info(f"CODE: {res.status_code} -- {res.reason}")
+        raise SystemExit()
+    change_log_json = res.json().get("changeLogJson")
+    if not change_log_json:
+        logger.info("No changeLogJson element is found!")
+        raise SystemExit()
+    digest = change_log_json.get("to").get("digest")
+    # if there's no digest found it means the part of the system responsible checks for updated content and if
+    # so recycles the underlying pods.  It takes roughly 8 mins for the git-cache to fully recycle.
+    # Let's do a brute force try for every 30 seconds.
+    max_retries = 16
+    sleep_time = 30
+
+    if not digest:
+        if current_try == max_retries:
+            logger.error(f"Imgage digest failed to show after {max_retries} retries!")
+            return
+        else:
+            logger.debug(f"Current try: {current_try}")
+            logger.warn(
+                f"E001: No image digest found, going to retry after {sleep_time} seconds"
+            )
+            time.sleep(sleep_time)
+            get_image_digest(url, current_try + 1)
+    else:
+        logger.info(f"Image Digest: {digest}")
+    return digest
+
+
 @click.command()
 @click.pass_context
 def image_signed_check(ctx):
     """
     Check payload image is well signed
     """
-    cs = ctx.obj["cs"] 
+    cs = ctx.obj["cs"]
     report = WorksheetManager(cs).get_test_report()
     image_signed_check_result = report.get_task_status(LABEL_TASK_PAYLOAD_IMAGE_VERIFY)
     if image_signed_check_result == TASK_STATUS_PASS:
         logger.info("image signed check already pass, not need to trigger again")
     else:
-        report.update_task_status(LABEL_TASK_PAYLOAD_IMAGE_VERIFY, TASK_STATUS_INPROGRESS)
+        report.update_task_status(
+            LABEL_TASK_PAYLOAD_IMAGE_VERIFY, TASK_STATUS_INPROGRESS
+        )
         try:
-            res = requests.get(cs.get_release_url() + "/releasestream/4-stable/release/" + cs.release)
-            res.raise_for_status()
-            soup = BeautifulSoup(res.content, 'lxml')
```

Comment:
if we won't use library `beautifulsoup4` and `lxml`, we need to remove them from file https://github.com/openshift/release-tests/blob/master/pyproject.toml